resources/design-document.md

```diff
@@ -1,75 +1,222 @@
+# 6. API
 
-_(You should have a separate section for each of the endpoints you are expecting to build...)_
+## 6.1. Public Models
+```
+//WarehouseModel
+String userId 
+String wareHouseId
+String name
+int region
+Map<ItemModel, Integer> inventory
+```
+
+```
+//ItemModel
+String itemId
+String name
+String category
+String regionOfOrigin
+int regionalDemand
+double salesForecast
+double perPallet
+double weight
+double purchaseCost
+double baseMargin
+double rateOfReplenishment
+String synergy
+boolean active
+```
+
+```
+//TransactionModel
+transactionId
+String warehouseId
+String shipmentId
+String itemId
+int count
+Date date
+String partnerId
+String transactionType
+
+```
+
+## 6.2. Create Item Endpoint
+
+* Accepts POST requests to /items
+* Accepts input data and user info to create a new item and returns an itemModel for the logged-in user
+
+## 6.3 Get Item Endpoint
+* Accepts GET requests to /items/:itemId
+* Accepts an itemId and user info and returns the relevant itemModel
+
+## 6.4 Get All Items Endpoint
+* Accepts GET requests to /items
+* Accepts user info and return a list of all Items
+
+## 6.5 Update Item Endpoint
+* Accepts PUT requests to /items/:itemId
+* Accepts an itemId, input data, and user info, updates the item data, and returns an itemModel
+
+## 6.6 Create Warehouse Endpoint
+* Accepts POST requests to /warehouses
+* Accepts input data and user info to create a new warehouse
+* Returns the created warehouseModel
+
+## 6.7 Get Warehouse Endpoint
+* Accepts GET requests to /warehouses/:warehouseId
+* Accepts a warehouseId and user info, and returns the relevant warehouseModel
+
+## 6.8 Get All Warehouses Endpoint
+* Accepts GET requests to /warehouses
+* Accepts user information and returns all warehouses associated with the account
+
+## 6.9 Delete Warehouse Endpoint
+* Accepts DELETE requests to /warehouses/warehouseId
+* Accepts a warehouseId and user info, and deletes the matching warehouse, and shipping data
+
+## 6.10 Update Warehouse Endpoint
+* Accepts PUT requests to /warehouses/:warehouseId
+* Accepts a warehouseId, allows the user to change the warehouse name, and returns the warehouseModel
+* 
+## 6.11 Update Warehouse Inventory Endpoint
+* Accepts PUT requests to /warehouses/inventory/:warehouseId
+* Accepts a warehouseId, itemIds in a map of stock change amounts for each item, a partnerID, an optional date field, and user info
+* Generates a transactionModel object and adds it to the transactions table for reporting
+* Updates warehouse inventory count and returns warehouseModel
```

Comment:
For consistency/predictability, what do you think `/warehouse/:warehouseId/inventory` as the URI (rather than swapping the order relative to other warehouse endpoints)?

Reply:
That URI change sounds good to me.  My thinking was that a variable section coming after static sections seemed more appropriate somehow, but that doesn't really hold water on second thought.